File: ruby/soap_translate.rb
require 'soap/wsdlDriver'
require 'google/cloud/translate'

wsdl = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL'
client = SOAP::WSDLDriverFactory.new(wsdl).create_rpc_driver

n = ARGV[0] || '10'
response = client.NumberToWords('ubiNum' => n.to_i)
english = response.NumberToWordsResult

translate = Google::Cloud::Translate.translation_v2_service do |config|
  config.credentials = ENV['GOOGLE_APPLICATION_CREDENTIALS'] if ENV['GOOGLE_APPLICATION_CREDENTIALS']
end
spanish = translate.translate(english, to: 'es').text
puts spanish

File: node/soap_translate.js
import soap from 'soap';
import translate from '@vitalets/google-translate-api';

const wsdl = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';
const n = process.argv[2] || '10';

soap.createClient(wsdl, (err, client) => {
  if (err) throw err;
  client.NumberToWords({ ubiNum: parseInt(n, 10) }, async (err, result) => {
    if (err) throw err;
    const english = result.NumberToWordsResult;
    const res = await translate(english, { to: 'es' });
    console.log(res.text);
  });
});
